src/ext/request_hooks.c

```diff
@@ -68,6 +68,54 @@ int dd_no_blacklisted_modules(TSRMLS_D) {
     return no_blacklisted_modules;
 }
 
+static inline BOOL_T _starts_with(const char *with, const char *str, size_t str_len) {
+    size_t with_len = strlen(with);
+    if (str_len < with_len) {
+        return FALSE;
+    } else {
+        return memcmp(with, str, with_len) == 0;
+    }
+}
+
+static inline void _resolve_path(const char *path, size_t path_len, const char *token, char *resolved_path) {
+    size_t length = path_len + strlen(token) + 1;
+    char *tmp_path = calloc(length, 1);
+    int last_slash = strrchr(path, '/') - path;
+
+    snprintf(tmp_path, length, "%.*s%s", last_slash, path, token);
+    realpath(tmp_path, resolved_path);
+    free(tmp_path);
+}
+
+char *ddtrace_fetch_computed_request_init_hook_path() {
+    Dl_info dl_info;
+    if (dladdr(ddtrace_fetch_computed_request_init_hook_path, &dl_info) == -1) {
+        return NULL;
+    }
+    char *path = (char *)dl_info.dli_fname;
+
+    if (access(path, F_OK) == -1) {
+        return NULL;
+    }
+    size_t path_len = strlen(path);
+    char *resolved_path = calloc(PATH_MAX + 1, 1);
```

Comment:
IDK, depends on how we approach this. 
In general the approach I take is that if a function returns a pointer - the callee responsibility to free it. But before that function exits the pointer is its own, if it decides to free it and return NULL thats ok too, because no ownership is transfer this way.

I think its a somewhat common pattern, at least its been used a lot here.
But from a POV of counting mallocs and free's  (eyeballing if everything what was allocated is freed) your approach makes much more sense if both are in the same place.

one thing I don't like here, is that we need to guess if the data needs to be freed using free or efree 🤔.

Reply:
I moved all the code to request_hook.c
there started to be way too many things in the RINIT. Also most importantly it hid a bug(s) when RINIT hook would finish early when IonCube module was found.